### PR TITLE
Consolidate all alg notes into a single clause note

### LINKED ISSUE #26: Duplication of Note in Evaluation
In the Evaluation clause, the existing note is mostly-duplicated 5 times (twice for the existing algorithms and 3 times for the new ones). I think it would be better to retain just one note and reword it to apply to all 5 algorithms. (E.g., don't refer to specific step numbers, just refer to the invocation of `PutValue` that occurs in each algorithm.)

### PR DESCRIPTION
**Pull request by jridgewell**
Re: https://github.com/tc39/proposal-logical-assignment/issues/17#issuecomment-602913774
Fixes #26

/cc @jmdyck and @syg

### CODE REVIEW

**Comment by anba** [comment]
The note is already incorrect in the main spec, so while you're here, maybe fix the step reference, too?  (Step 7 for compound assignment won't throw a `ReferenceError` for unresolvable references in strict mode, because `GetValue(lref)` in step 2 already throws.)

**Comment by jridgewell** [comment]
Done.
